nodeinfo/manager.go

```diff
@@ -48,13 +57,15 @@ type (
 
 	// InfoManager manage delegate node info
 	InfoManager struct {
-		version       string
-		address       string
-		nodeMap       *lru.Cache
-		broadcastTask *routine.RecurringTask
-		transmitter   transmitter
-		chain         chain
-		privKey       crypto.PrivateKey
+		lifecycle.Lifecycle
+		version      string
+		address      string
+		isDelegate   atomic.Value // bool
```

Comment:
why should `atomic.value` be used here?

Reply:
Forget to restore the type to `bool` from `atomic.Value` after removing another goroutine to update it